pkg/workloadmeta/store.go

```diff
@@ -275,6 +275,48 @@ func (s *store) GetKubernetesPod(id string) (*KubernetesPod, error) {
 	return entity.(*KubernetesPod), nil
 }
 
+// GetProcess implements Store#GetProcess.
+func (s *store) GetProcess(pid int32) (*Process, error) {
+	id := string(pid)
+
+	entity, err := s.getEntityByKind(KindProcess, id)
+	if err != nil {
+		return nil, err
+	}
+
+	return entity.(*Process), nil
+}
+
+// ListProcesses implements Store#ListProcesses.
+func (s *store) ListProcesses() []*Process {
+	entities := s.listEntitiesByKind(KindProcess)
+
+	// Not very efficient
+	processes := make([]*Process, 0, len(entities))
+	for i := range entities {
+		processes = append(processes, entities[i].(*Process))
```

Comment:
don't use append, when u can assign by index

Reply:
Given that the slice is already pre-allocated with the right capacity, is there any difference regarding performance between assigning by index or using append? Or is it more of a stylistic concern?

Reply:
I think both are equivalent. Internally, `Append` copies the whole slice if it needs to increase the capacity. If the capacity does not need to be increased then it should not change anything.